ext/standard/http_fopen_wrapper.c

```diff
@@ -369,8 +373,8 @@ static php_stream *php_stream_url_wrap_http_ex(php_stream_wrapper *wrapper,
 			/* As per the RFC, automatically redirected requests MUST NOT use other methods than
 			 * GET and HEAD unless it can be confirmed by the user */
 			if (!redirected
-				|| (Z_STRLEN_P(tmpzval) == 3 && memcmp("GET", Z_STRVAL_P(tmpzval), 3) == 0)
-				|| (Z_STRLEN_P(tmpzval) == 4 && memcmp("HEAD",Z_STRVAL_P(tmpzval), 4) == 0)
+				|| zend_string_equals_literal_ci(Z_STR_P(tmpzval), "GET")
+				|| zend_string_equals_literal_ci(Z_STR_P(tmpzval), "HEAD")
```

Comment:
The previous comparison was case-sensitive.

Reply:
too careless 😵